### PR TITLE
upgrade ruby-maven-libs to 3.8.9

### PR DESCRIPTION
**Pull request by jsvd**
~~Given Logstash doesn't ship with any gems that download jars at install time we can exclude the ruby maven lib jars~~

Upgrade ruby-maven-libs to 3.8.9

### CODE REVIEW

**Comment by jsvd** [comment]
Unfortunately this does break plugin installation in the case where jars need to be downloaded at install time. For example:

```
❯ bin/logstash-plugin install logstash-output-oss
Using bundled JDK: /Users/joaoduarte/elastic/logstash/build/logstash-8.14.0-SNAPSHOT/jdk.app/Contents/Home
Validating logstash-output-oss
Resolving mixin dependencies
WARN: Unresolved or ambiguous specs during Gem::Specification.reset:
      date (>= 0)
      Available/installed versions of this gem:
      - 3.3.4
      - 3.3.3
WARN: Clearing out unresolved specs. Try 'gem cleanup <gem>'
Please report a bug if this causes problems.
Installing logstash-output-oss
java.lang.ClassNotFoundException: org.apache.maven.cli.MavenCli
	at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass(SelfFirstStrategy.java:50)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass(ClassRealm.java:271)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:247)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:239)
	at org.codehaus.plexus.classworlds.launcher.Launcher.getMainClass(Launcher.java:144)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:266)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Error Errno::ENOENT, retrying 1/10
No such file or directory - /Users/joaoduarte/elastic/logstash/build/logstash-8.14.0-SNAPSHOT/vendor/bundle/jruby/3.1.0/gems/logstash-output-oss-0.1.2-java/deps.lst
Installation successful
```

**Comment by jsvd** [comment]
With ruby-maven-libs 3.8.9 out, I've repurposed this plugin to upgrade the gem instead of removing the jars from it.

File: rakelib/artifacts.rake [hunk 1]
```diff
@@ -99,6 +99,8 @@ namespace "artifact" do
     @exclude_paths << 'vendor/**/gems/**/Gemfile'
 
     @exclude_paths << 'vendor/jruby/lib/ruby/gems/shared/gems/rake-*'
+    # exclude the maven jars until https://github.com/jruby/ruby-maven-libs/pull/4
+    @exclude_paths << 'vendor/bundle/jruby/**/gems/ruby-maven-libs-3.3.9'
```
Comment (andsel) [inline rakelib/artifacts.rake:103]: Why exclude `ruby-maven-libs-3.3.9` is it a typo and you meant `3.8.9` or this `3.3.9` are kicked in by something else?
In my local (in `vendor/bundle/jruby/3.1.0/gems/` ) I can see it contains:
```
ruby-maven-libs-3.3.9/  
ruby-maven-libs-3.8.9/
```

So I suspect  the 3.3.9 is undesired but not clear what picks it

**Review by andsel — COMMENTED** [review]
Left a comment asking for more clarification

File: rakelib/artifacts.rake (hunk 1 above)
Comment (jsvd) [reply rakelib/artifacts.rake:103]: The goal is to not include the version that comes with JRuby. And from what I saw the files for the older version were still being left behind after the upgrade. This ensures the older version is not in the artifact.

**Comment by shreesaumya** [comment]
Logstash 8.13 version is still having two version of ruby-maven-libs, ruby-maven-libs-3.3.9/ and ruby-maven-libs-3.8.9/. Are both dependencies needed. ?

A customer posed this question in SF case [01602194](https://elastic.lightning.force.com/lightning/r/Case/500Vv000004J5XxIAK/view?ws=%2Flightning%2Fr%2FReport%2F00O8X000008gKcjUAE%2Fview%3FreportFilters%3D%255B%257B%2522operator%2522%253A%2522equals%2522%252C%2522value%2522%253A%2522Saumya%2520Shree%2522%252C%2522column%2522%253A%2522OWNER%2522%257D%255D)

**Pull request merged**